[MP Rewrite Part 6] Cleanup and fixes

kylestach opened the pull request:
This is a collection of assorted fixes and cleanup required to make the previous sets of commits pass CI. I chose to avoid putting them directly into the correct commits for timing reasons, it wouldn't have allowed them to pass CI anyway.

kylestach committed:
Cleanup and fixes to make code run properly

kylestach committed:
Fix test compilation and disable a trapezoidal motion profile edge case.

Some planner tests are still failing. This will be addressed in a later
commit.

kylestach committed:
Prettify

kylestach committed:
Make e2e planner tests pass.

Because they generate random scenarios, and the RRT planner is not
guaranteed to converge in a finite number of iterations (and the Bezier
approximation may be incorrect), these tests are currently flaky. That
should be fixed before this is merged to staging.

kylestach committed:
Tidy fixes

kylestach committed:
Linting

kylestach committed:
Formatting

oswinso commented on soccer/planning/tests/PlannerTest.cpp:
```diff
@@ -44,8 +48,31 @@ TEST(Planning, path_target_random) {
+
+        if (!checkPathContinuous(path, RobotConstraints{})) {
+            std::cout << "Saving to /tmp/out.csv" << std::endl;
+            std::ofstream file("/tmp/out.csv");
+            for (auto cursor = path.cursor_begin(); cursor.has_value();
+                 cursor.advance(0.05s)) {
+                auto instant = cursor.value();
+                file << RJ::Seconds(instant.stamp - RJ::now()) << ", "
+                     << instant.position().x() << ", " << instant.position().y()
+                     << ", " << instant.linear_velocity().x() << ", "
+                     << instant.linear_velocity().y() << std::endl;
```
What's this file for?

kylestach replied:
Basically just smoke testing the planners, but I had it writing to a file for debugging purposes. I'll go through and clean all of this out.

kylestach replied:
Tests will be fixed in a later PR to mp-rewrite/base.

JNeiger commented on soccer/config/sim.xml:
```diff
@@ -358,38 +355,40 @@
+  <searchIncDist value="0.2"/>
+  <maxBallSpeedDirect value="0.1"/>
+  <maxApproachAngle value="0.3"/>
+  <maxOutrunBallSpeed value="0.7"/>
```
Some of these don't exist?

kylestach replied:
Fixed up all of the configs to match staging and cleared out unused ones.

JNeiger commented on soccer/config/sim.xml:
```diff
@@ -358,38 +355,40 @@
-   <ballVelGain value="0.9"/>
-   <maxAnglePathTargetChange value="20"/>
-   <maxBallVelForPathReset value="2"/>
-   <maxBallAngleForReset value="20"/>
```
A bunch of these have been deleted reverting to their default value. At one point I set the defaults to my best config at the time, but it's been tuned since them I imagine

kylestach committed:
Merge branch 'mp-rewrite/planner-ports' into mp-rewrite/cleanup-and-fixes

kylestach committed:
Remove deleted file from CMakeLists, remove reuse condition from settle

kylestach committed:
Don't replan in Collect::control

kylestach committed:
Merge branch 'mp-rewrite/planner-ports' into mp-rewrite/cleanup-and-fixes

kylestach committed:
Optimizations to geometry methods

kylestach committed:
Reset and regenerate sim config

This deletes unnecessary variables and moves planning-related variables
into PathPlanner

kylestach committed:
Formatting

JNeiger commented on soccer/planning/planner/EscapeObstaclesPathPlanner.cpp:
```diff
@@ -31,12 +32,17 @@ Trajectory EscapeObstaclesPathPlanner::plan(const PlanRequest& planRequest) {
 
-    if (!obstacles.hit(startInstant.position())) {
-        // Stop in place
+    // TODO(#
```
lost the todo number

kylestach replied:
Yeah, oops. This TODO actually shouldn't exist, it's fine for now.

kylestach committed:
Add TODO in EscapeObstacles

kylestach committed:
Tidy fixes

kylestach committed:
Revert inconsistent clang-tidy recommended fix

kylestach committed:
Temporarily disable flaky planner test

kylestach commented:
Just a heads up that this is not the last PR in the motion planning rewrite as a whole, just the last in this set. I am planning on squashing it all down into the base PR and making more PRs to improve test coverage and make a few more bugfixes/improvements.

kylestach commented:
Sigh...the clang-tidy failure is really stupid (and in fact their suggested fix gives the obvious unnecessary `std::move` warning on my machine):
```
/root/robocup-software/soccer/Robot.cpp:207:46: warning: parameter 'target' is passed by value and only copied once; consider moving it to avoid unnecessary copies [performance-unnecessary-value-param]
    setMotionCommand(Planning::SettleCommand{target});
                                             ^
                                             std::move( )
```

oswinso commented:
Just NOLINT it then probably

kylestach commented:
Yeah, I just really hate adding NOLINTs just to satisfy the linter. It dilutes actually valid NOLINTs, especially because NOLINTS never actually end up getting removed.

kylestach committed:
Add NOLINTNEXTLINE

oswinso commented on soccer/config/sim2.xml:
```diff
@@ -0,0 +1,408 @@
+<config>
```
What's this file for?

kylestach replied:
Crap.

Will remove lmao

oswinso commented on soccer/planning/TrajectoryUtils.cpp:
```diff
@@ -79,11 +75,11 @@ bool TrajectoryHitsDynamic(const Trajectory& trajectory,
     // something is probably wrong, but we'll still handle it (just scale dt
     // accordingly)
-    constexpr int max_iterations = 400;
+    constexpr int max_iterations = 100;
```
I feel like this should be a parameter somewhere? Maybe todo or something?

kylestach replied:
Eh, this isn't really something we'll ever want to change...it doesn't affect behavior, it should basically be "as low as possible without affecting behavior". It can be a param though, sure.

oswinso commented on soccer/planning/tests/PlannerTest.cpp:
```diff
@@ -24,7 +24,7 @@ using namespace Planning;
 using namespace Planning::TestingUtils;
 
-TEST(Planning, path_target_random) {
+TEST(Planning, DISABLED_path_target_random) {
```
Why is this disabled?

kylestach replied:
Flaky tests.

kylestach replied:
I'll make it deterministic in a later PR.

kylestach committed:
Remove old sim config and add TODOs for trajectory utils

oswinso reviewed (approved):
lgtm
